File: Arista.java
public class Arista {
    // Configuración global ajustable
    private static int MAX_VEH_PER_MIN = 100;
    private static float MULT_CALLE = 1.20f;
    private static float MULT_AVENIDA = 0.80f;

    public static int getMaxVehPerMin() { return MAX_VEH_PER_MIN; }
    public static void setMaxVehPerMin(int v) { MAX_VEH_PER_MIN = Math.max(1, v); }

    public static float getMultCalle() { return MULT_CALLE; }
    public static void setMultCalle(float m) { MULT_CALLE = Math.max(0f, m); }

    public static float getMultAvenida() { return MULT_AVENIDA; }
    public static void setMultAvenida(float m) { MULT_AVENIDA = Math.max(0f, m); }

    // Estado de instancia
    protected Interseccion from;
    protected Interseccion to;
    protected float weight; // 0..1
    protected int vehiculosPorMin;
    protected float weightMultiplier = 1.0f;

    public Arista(Interseccion f, Interseccion t) {
        this.from = f;
        this.to = t;
        this.weight = 0f;
        this.vehiculosPorMin = 0;
    }

    // Compatibilidad: delega al nuevo cálculo
    public void calcularPeso() { recalcPeso(); }

    public void setVehiculosPorMin(int vpm) {
        if (vpm < 0) vpm = 0;
        this.vehiculosPorMin = vpm;
        recalcPeso();
    }

    public int getVehiculosPorMin() { return this.vehiculosPorMin; }
    public float getWeight() { return this.weight; }
    public Interseccion getFrom() { return this.from; }
    public Interseccion getTo() { return this.to; }
    public float getWeightMultiplier() { return this.weightMultiplier; }

    protected void setWeightMultiplier(float m) {
        this.weightMultiplier = Math.max(0f, m);
        recalcPeso();
    }

    public Interseccion getOrigen() { return this.from; }
    public Interseccion getDestino() { return this.to; }

    protected void recalcPeso() {
        float base = (float) this.vehiculosPorMin / (float) MAX_VEH_PER_MIN;
        float raw = base * this.weightMultiplier;
        if (raw < 0f) raw = 0f;
        if (raw > 1f) raw = 1f;
        this.weight = raw;
    }

    public void onConfigChanged() { recalcPeso(); }

    @Override
    public String toString() {
        return "Arista[" + from + " -> " + to + ", veh/min=" + vehiculosPorMin + ", peso=" + String.format("%.3f", weight) + "]";
    }
}

File: Arista.js
/**
 * Clase Arista - Representa una conexión entre dos intersecciones
 * Equivalente a Arista.java
 */
export class Arista {
    static MAX_VEH_PER_MIN = 100;
    static TIPO_CALLE = 'calle';
    static TIPO_AVENIDA = 'avenida';

    constructor(from, to, tipo) {
        this.from = from; // Interseccion
        this.to = to;     // Interseccion
        this.tipo = tipo; // 'calle' o 'avenida'
        this.vehiculosPorMin = 0;
        this.weight = 0;
        this.weightMultiplier = 1.0;
    }

    /**
     * Calcula el peso de la arista basado en el tráfico
     * Replica calcularPeso() en Java
     */
    calcularPeso() {
        let rawWeight = this.vehiculosPorMin / Arista.MAX_VEH_PER_MIN;
        rawWeight *= this.weightMultiplier;
        
        // Clampear entre 0 y 1
        this.weight = Math.max(0, Math.min(1, rawWeight));
        
        return this.weight;
    }

    /**
     * Establece el tráfico en vehículos por minuto
     */
    setVehiculosPorMin(vehiculos) {
        this.vehiculosPorMin = Math.max(0, Math.min(Arista.MAX_VEH_PER_MIN, vehiculos));
        this.calcularPeso();
    }

    /**
     * Establece el multiplicador de peso
     */
    setWeightMultiplier(multiplier) {
        this.weightMultiplier = Math.max(0, multiplier);
        this.calcularPeso();
    }

    /**
     * Obtiene el identificador único de la arista
     */
    getId() {
        return `${this.from.getId()}->${this.to.getId()}`;
    }

    /**
     * Obtiene el nivel de congestión como porcentaje
     */
    getNivelCongestion() {
        return Math.round(this.weight * 100);
    }

    /**
     * Obtiene el color basado en el nivel de congestión
     */
    getColor() {
        if (this.weight < 0.3) return '#4ade80'; // Verde (bajo)
        if (this.weight < 0.6) return '#facc15'; // Amarillo (medio)
        if (this.weight < 0.8) return '#fb923c'; // Naranja (alto)
        return '#ef4444'; // Rojo (crítico)
    }

    /**
     * Representación en string de la arista
     */
    toString() {
        return `Arista[${this.tipo}](${this.from.toString()} -> ${this.to.toString()}, peso: ${this.weight.toFixed(3)})`;
    }

    /**
     * Clona la arista
     */
    clone() {
        const cloned = new Arista(this.from, this.to, this.tipo);
        cloned.vehiculosPorMin = this.vehiculosPorMin;
        cloned.weightMultiplier = this.weightMultiplier;
        cloned.calcularPeso();
        return cloned;
    }

    /**
     * Serializa la arista a JSON
     */
    toJSON() {
        return {
            from: this.from.getId(),
            to: this.to.getId(),
            tipo: this.tipo,
            vehiculosPorMin: this.vehiculosPorMin,
            weightMultiplier: this.weightMultiplier,
            weight: this.weight
        };
    }
}